src/solana/rpc/api.py

```diff
@@ -428,48 +339,14 @@ def get_blocks(self, start_slot: int, end_slot: Optional[int] = None) -> types.R
             >>> solana_client.get_blocks(5, 10) # doctest: +SKIP
             {'jsonrpc': '2.0', 'result': [5, 6, 7, 8, 9, 10], 'id': 1}
         """
-        args = self._get_blocks_args(start_slot, end_slot)
-        return self._provider.make_request(*args)
-
-    def get_confirmed_signature_for_address2(
-        self,
-        account: Union[str, Keypair, PublicKey],
-        before: Optional[str] = None,
-        until: Optional[str] = None,
-        limit: Optional[int] = None,
-        commitment: Optional[Commitment] = None,
-    ) -> types.RPCResponse:
-        """Returns confirmed signatures for transactions involving an address.
-
-        Signatures are returned backwards in time from the provided signature or
-        most recent confirmed block.
-
-        Args:
-            account: Account to be queried.
-            before: (optional) Start searching backwards from this transaction signature.
-                If not provided the search starts from the top of the highest max confirmed block.
-            until: (optional) Search until this transaction signature, if found before limit reached.
-            limit: (optional) Maximum transaction signatures to return (between 1 and 1,000, default: 1,000).
-            commitment: (optional) Bank state to query. It can be either "finalized", "confirmed" or "processed".
-
-        Example:
-            >>> solana_client = Client("http://localhost:8899")
-            >>> solana_client.get_confirmed_signature_for_address2("Vote111111111111111111111111111111111111111", limit=1) # doctest: +SKIP
-            {'jsonrpc': '2.0',
-             'result': [{'err': None,
-               'memo': None,
-               'signature': 'v1BK8XcaPBzAGd7TB1K53pMdi6TBGe5CLCgx8cmZ4Bj63ZNvA6ca2QaxFpBFdvmpoFQ51VorBjifkBGLTDhwpqN',
-               'slot': 4290}],
-            'id': 2}
-        """  # noqa: E501 # pylint: disable=line-too-long
-        args = self._get_confirmed_signature_for_address2_args(account, before, until, limit, commitment)
-        return self._provider.make_request(*args)
+        body = self._get_blocks_body(start_slot, end_slot)
+        return self._provider.make_request(body)
 
     def get_signatures_for_address(
         self,
-        account: Union[str, Keypair, PublicKey],
-        before: Optional[str] = None,
-        until: Optional[str] = None,
+        account: PublicKey,
+        before: Optional[Signature] = None,
```

Comment:
Can we use an internal helper class/method that converts `str` to https://github.com/kevinheavey/solders/blob/fdb1ce9beac0372a42be4af34326f51a48bf55aa/python/solders/signature.pyi#L4, instead of exposing it directly through the client?

Reply:
Can `from_string` be used here?

Reply:
I prefer the `Signature` object for two reasons:

1. It's more self-documenting (you don't need to read the docs to find you need a base58-encoded string).
2. I'm working on a PR (for the same release as this one) to add RPC response parsing, and that will parse the `sendTransaction` response into a `Signature`. So the most common source of signatures will already be a `Signature` object.